src/hostname/hostnamed.c

```diff
@@ -822,6 +822,30 @@ static int property_get_uname_field(
         return sd_bus_message_append(reply, "s", (char*) &u + PTR_TO_SIZE(userdata));
 }
 
+static int get_bios_version(char **ret) {
```

Comment:
hmm, any chance you can generalize the existing get_hardware_serial() call a bit for this?

i.e. make it accept the sysattr name to read as additional param (or two, as get_hardware_serial() wants that). and then just provide two tiny wrappers around this generalization. one that passes the strings "product_serial"/"board_serial", and the other one: "bios_version".

get_dmi_data() does something similar, but for device properties instead of attributes. But it can be used as a template here.

(in case you wonder: udev device properties are uppercase fields assigned by udev in userspace to a device, and sysfs device attributes are lowercase fields exposed by the kernel directly in sysfs. The former might be based on the latter though. Confusin? yes! :-))

Reply:
Yes. Updated the get_hardware_serial as get_hardware_firmware_version that passes strings "product_serial"/"board_serial", and the other one: "bios_version". and returns accordingly